src/marqo/core/models/marqo_index.py

```diff
@@ -302,10 +309,15 @@ def lexically_searchable_fields_names(self) -> Set[str]:
 
     @property
     def filterable_fields_names(self) -> Set[str]:
-        return self._cache_or_get('filterable_fields_names',
-                                  lambda: {field.name for field in self.fields if
-                                           FieldFeature.Filter in field.features}
-                                  )
+        def generate():
+            the_set = {field.name for field in self.fields if
+                       FieldFeature.Filter in field.features}
+            if self.parsed_marqo_version().match('>=2.2.0'):
```

Comment:
Is it necessary to mock this version of `parsed_marqo_version()` and add a unit test on this?